### PR TITLE
Add documentation for new 1.8.1 features + add new option in config - update interval

### PR DESCRIPTION
**Pull request by mkz212**
- add documentation
- change default refresh interval from 60 s to 10 s
- add option in config - Refresh Interval

### CODE REVIEW

**Comment by embee8** [comment]
Version increase missing in `package-lock.json`. Please update in line with `package.json`, as per process in CONTRIBUTING.md.

**Comment by embee8** [comment]
Feedback on the commit messages: Please make them more descriptive going forward so that reviewers know what was changed in a commit at a glance. A decent resource on the topic is: https://www.freecodecamp.org/news/writing-good-commit-messages-a-practical-guide.

The pattern "Update FILE" isn't very descriptive.

image

File: package.json
Comment (embee8) [inline package.json:4]: Needs to be "1.8.1".

File: package-lock.json
Comment (embee8) [inline package-lock.json:3]: Needs to be "1.8.1".

There is also a change missing here, which results from updating version numbers manually and not using the `npm version` command that can be found in the development guide. Either switch to using the command or complete the full change manually.

**Review by embee8 — CHANGES_REQUESTED** [review]
See added comments around version number increase.

File: package-lock.json
Comment (jandersonhill) [reply package-lock.json:3]: @embee8 If I were to run npm version command on behalf of @mkz212, should I edit package-lock.json back to '1.8.0' first?

Sacrilege, I know, but what would be the result of editing both package.json and package-lock.json to the new version number by hand and then switching to using 'npm version' thereafter? I agree it's not ideal, but it would resolve trying to learn too many things at the same time, while still allowing the beta to progress.

Comment (embee8) [reply package-lock.json:3]: Version 1.8.0 has already been published as beta, so it cannot be re-published as beta. If you would change the version back to 1.8.0 and run the production release workflow, the beta will turn into a production release. It won't even re-release and overwrite the old version with the new changes.

The process is set up in a way that you introduce new releases as beta and then promote them to production. If you publish a package with a version like "1.8.0-beta.2", this workflow kind of breaks, at least semantically.

As a rule of thumb: The only way is forward. Every time new things are added, increase the version number, release as beta. If you are happy with the outcome, run the production workflow for the same version.

Sorry for the complexity added with these new workflows. But you asked for the ability to run betas, and there is no such thing as a free lunch. 😬

Comment (jandersonhill) [reply package-lock.json:3]: Sorry, I probably wasn't very clear. I believe I/we have two options with versioning for the next beta release:

Option 1)

The better option is to run 'npm version' to create a new beta version number which would modify the necessary files, but do I need to change package-lock.json back to 1.8.0 before running this command?

Option 2)

The less good option is if I changed package-lock.json and package.json manually, but would this then break the use of 'npm version' on the next release or would it pick up the version number from the package*.json files?

That said, it seems you've thought this through, and I may just have to bite the bullet and get to grips with npm before doing anything else :-)

Comment (embee8) [reply package-lock.json:3]: Ah, yes. Those are the two options you have. I encourage you to just experiment with the `npm version` command. You can always revert the changes or simply not commit them, or close the pull request and not merge the changes. You will see that the command changes the version in three places if I remember correctly – those should always be aligned.

Comment (jandersonhill) [reply package-lock.json:3]: I appreciate the help :-) It might take me a couple of evenings, but I'll figure out npm, look at this and the token.

Comment (jandersonhill) [reply package-lock.json:3]: @mkz212 Wanted to make sure you're aware of the conversation above. I can resolve the versioning if you like, but give me a couple of days to get to grips with npm (I'm not around much tomorrow).

Comment (embee8) [reply package-lock.json:3]: Probably easiest if @mkz212 makes the change on his already established local branch and pushes the commit to the remote which will update this PR accordingly.

Comment (mkz212) [reply package-lock.json:3]: > Version 1.8.0 has already been published as beta, so it cannot be re-published as beta. If you would change the version back to 1.8.0 and run the production release workflow, the beta will turn into a production release. It won't even re-release and overwrite the old version with the new changes.
> 
> The process is set up in a way that you introduce new releases as beta and then promote them to production. If you publish a package with a version like "1.8.0-beta.2", this workflow kind of breaks, at least semantically.
> 
> As a rule of thumb: The only way is forward. Every time new things are added, increase the version number, release as beta. If you are happy with the outcome, run the production workflow for the same version.
> 
> Sorry for the complexity added with these new workflows. But you asked for the ability to run betas, and there is no such thing as a free lunch. 😬

IMG_4486
IMG_4485

Comment (mkz212) [reply package-lock.json:3]: @embee8 @jandersonhill Correct it as you want - I don't mind. 👍

Comment (embee8) [reply package-lock.json:3]: Regarding the screenshots and links above: It's the same process, only that the referenced case packs additional semantic information into the version number/string itself, whereas I had in mind to use dist-tags for that.

Either approach works, it should only be applied consistently and most importantly play well with how Homebridge handles plugin updates. Let me study the Homebridge core code on this and come back to you.

Comment (mkz212) [reply package-lock.json:3]: @Embee8 in every plugin I worked with always the public version was like 1.5.0 to 1.5.1 or 1.5.0 to 1.6.0 etc. But the beta version has always been in the style of 1.5.0-beta.0 to 1.5.0-beta.1 etc. Recently in Config Ui it came to beta.97 - you imagine as if the start was, for example, with 1.5.0, with 97 beta versions where it would end - it would be a real chaos.

Another thing - I didn't meet anywhere so that the PR reporter had to decide on the choice of version. Many creators simply run npm version before running the npm publish command. So maybe instead of the function checking if there is a version number, just give the function that increases the version number accordingly? I think it will be easier and better.

Comment (embee8) [reply package-lock.json:3]: homebridge-config-ui-x uses a different / continuous deployment model, as you can see in how [frequently new versions are released](https://www.npmjs.com/package/homebridge-config-ui-x?activeTab=versions). The comparison between a plugin that has 1,400+ versions through continuous deployment with this plugin which has 19 releases since its inception isn't particularly relevant at this point.

I studied the Homebridge [source code around plugin updates](https://github.com/homebridge/homebridge-config-ui-x/blob/f63405f68a55416be3f9bb3ee4d47227b78d691c/src/modules/plugins/plugins.service.ts) and it supports npm dist-tags.

> Another thing - I didn't meet anywhere so that the PR reporter had to decide on the choice of version. Many creators simply run npm version before running the npm publish command.

The current release process is designed to be light(er) touch for maintainers based on [previous discussions](https://github.com/homebridge-panasonic-ac-platform/homebridge-panasonic-ac-platform/issues/94). To support that, we ask contributors to help with tasks like version management, not least because they will have the most context to decide what kind of semver version change is appropriate based on the feature they introduce, change, or remove.

> So maybe instead of the function checking if there is a version number, just give the function that increases the version number accordingly? I think it will be easier and better.

It is doable, but I don't agree that it is easier. This function would either have to directly change the code through a new commit, or alternatively create a new PR for the version change which maintainers can review and merge. Compare this with the author of a PR simply including this in their packet of work.

That said, feel free to change this to any process or format that optimally supports the maintenance of the plugin. As previously stated, I don't have the bandwidth at the moment to pursue this, so don't feel limited by my opinions.

Comment (mkz212) [reply package-lock.json:3]: 👍

Comment (jandersonhill) [reply package-lock.json]: @mkz212 I'm seeing an error message: 'This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository'. Are you able to assign a branch to this so I can commit? If not, let me know and I'll take a look.

This may be a reason @embee8 is suggesting we use npm for versioning.

Comment (mkz212) [reply package-lock.json]: I have no idea what it's about. If you know what's going on, please take care of it.

**Review by embee8 — APPROVED** [review]

**Comment by embee8** [comment]
@jandersonhill, the next steps here are to merge the pull request and then run the respective release workflow from the tag `v1.8.1` which will be created automatically once the PR is merged.

**Comment by jandersonhill** [comment]
Done!

**Comment by embee8** [comment]
Nice! Congratulations on your first release using the new workflow(s). 👏

**Comment by jandersonhill** [comment]
Thanks for all your help! Next on my list is to look into the npm token. The issues at home should be less demanding over the next few weeks, so things should run somewhat more smoothly now we're all used to the new workflow :-)

**Pull request merged**
